client/knowledge_base_search/src/containers/ResultsContainer/ResultComparison.js

```diff
@@ -32,49 +33,80 @@ class ResultComparison extends Component {
   }
 
   getFullAnswer() {
-    const { full_result_type, result, enriched_result } = this.props;
+    const { full_result_type, result, enriched_result, passages } = this.props;
 
-    return full_result_type === 'enriched' ? enriched_result : result;
+    switch(full_result_type) {
+      case 'enriched':
+        this.setState({full_answer: enriched_result});
+        break;
+      case 'passages':
+        Promise.all([
```

Comment:
you can just do `Promis.resolve(` when it is a single asynchronous operation -> see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve